File: programme/number-to-string/NumberToString.java
// Write a program to convert number to strings
import java.util.Scanner;

public class NumberToString {
    public static void main(String[] args) {
        String[] word = { "Zero", "One", "Two", "Three", "Four", "Five", "Six", "Seven", "Eight", "Nine" };
        Scanner sc = new Scanner(System.in);
        System.out.print("Input  : ");
        String a = sc.nextLine();
        System.out.print("Output : ");
        for (int i = 0; i < a.length(); i++) {
            System.out.print(word[(int) a.charAt(i) - 48] + " ");
        }
        System.out.println();
        sc.close();
    }
}

File: programme/number-to-string/number-to-string.go
// Write a programme to convert number to strings
package main

import (
	"fmt"
	"regexp"
)

var NumStrList = []string{"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"}

func main() {
	var str string

	fmt.Print("Input Number : ")
	fmt.Scan(&str)

	isNum, err := regexp.MatchString("^\\d+$", str)
	if err != nil || !isNum {
		fmt.Printf("Input error: %s", str)
		return
	}

	fmt.Print("\nOutput : ")
	for i := 0; i < len(str); i++ {
		fmt.Printf("%s ", NumStrList[str[i]-48]) // ascii code 48 -> 0
	}
	fmt.Println()
}
